src/api/scopes.js

```diff
@@ -0,0 +1,72 @@
+export class Scope {
+  scopes = [];
+
+  get finalQuery() {
+    return this.scopes.map((s) => `&${s.key}=${s.query}`).join("");
+  }
+
+  addScope = (key, query) => {
```

Comment:
Is there a reason you don't write
```suggestion
  addScope(key, query) {
```
?

Reply:
Not really, it shouldn't make any difference to write it one way or the other

Reply:
I like the `function() {}` syntax more, tbh.